File: Python-jk/136.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        d={}
        for num in nums:
            if num in d:
                d[num]=2
            else:
                d[num]=1
        for num in d:
            if d[num]==1:
                return num

File: C++/136.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        // 所有的值取异或，结果为只出现一次的值
        int res = 0;
        for(auto &x : nums)
            res ^= x;
        return res;
    }
};
